device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/MessageProperty.java

```diff
@@ -31,6 +31,11 @@
     public static final String IOTHUB_SECURITY_INTERFACE_ID = "iothub-interface-id";
     public static final String IOTHUB_SECURITY_INTERFACE_ID_VALUE = "urn:azureiot:Security:SecurityAgent:1";
 
+    //Note this list includes the characters for tab and space.
+    private static final String BANNED_NON_ALPHANUMERIC_CHARACTERS = "()<>@,;:\"/[]?={} \t";
```

Comment:
Is this a final set of characters? where did you get them from? is it defined somewhere? can we have a link as a comment?

Reply:
I got this requirement from our own documentation here: https://azure.github.io/azure-iot-sdk-java/device/com/microsoft/azure/iothub/MessageProperty.html